test/validation/api/ipsec/ipsec.c

```diff
@@ -1136,22 +1233,9 @@ int ipsec_init(odp_instance_t *inst, odp_ipsec_op_mode_t mode)
 		fprintf(stderr, "Packet pool creation failed.\n");
 		return -1;
 	}
-	if (mode == ODP_IPSEC_OP_MODE_ASYNC ||
-	    mode == ODP_IPSEC_OP_MODE_INLINE) {
-		out_queue = odp_queue_create("ipsec-out", NULL);
-		if (ODP_QUEUE_INVALID == out_queue) {
-			fprintf(stderr, "IPsec outq creation failed.\n");
-			return -1;
-		}
-
-		default_queue = odp_queue_create("ipsec-default", NULL);
-		if (ODP_QUEUE_INVALID == default_queue) {
-			fprintf(stderr, "IPsec defaultq creation failed.\n");
-			return -1;
-		}
-	}
 
-	if (mode == ODP_IPSEC_OP_MODE_INLINE) {
+	if (suite_context.inbound_op_mode == ODP_IPSEC_OP_MODE_INLINE ||
+	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_INLINE) {
```

Comment:
The mode argument removal is still here even though the patch was removed. Since the op mode fields of suite_context are not initialized before ipsec_init call, this code no longer works.

What I meant with the removal of the mode-removal patch was that no only the patch is removed but also the changes in it (i.e. not squashed to some other patch).

To fix the situation, I propose resurrecting the mode argument and undoing the changes related to it.

Reply:
Ack. Couldn't verify the tests locally before pushing due to some environment issues. Will bring back the mode arg related changes with next revision.